File: features/user/user_identification_common/src/main/java/com/tokopedia/user_identification_common/util/AppSchedulerProvider.java
package com.tokopedia.user_identification_common.util;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class AppSchedulerProvider implements SchedulerProvider {

    @Override
    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }

}

File: features/user/user_identification_common/src/main/java/com/tokopedia/user_identification_common/util/AppSchedulerProvider.kt
package com.tokopedia.user_identification_common.util

import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class AppSchedulerProvider : SchedulerProvider {
    override fun ui(): Scheduler? {
        return AndroidSchedulers.mainThread()
    }

    override fun io(): Scheduler? {
        return Schedulers.io()
    }
}